lib/takeaway.rb

```diff
@@ -0,0 +1,42 @@
+class Restaurant
+  
+  attr_reader :dishes_with_prices, :sum, :order
+  
+  def initialize
+    @dishes_with_prices = {
```

Comment:
Nice to use a Hash to store the menu information :) 
You could probably remove the instance variable `@sum` — because you can recalculate it from the `@order` array, since it contains the different dishes in the order, you could change the method `total` so it iterates on the `@order` array to recalculate the total, saving you from having to manage this `@sum` variable. As a good rule of thumb, any value that can be calculated or derived from another one shouldn't be stored